src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elementData = EMPTY_ELEMENT_DATA;
+    }
+
+    private void quickRemove(Object[] elements, int index) {
+        if ((size - 1) >= index) {
+            System.arraycopy(elements, index + 1, elements, index, (size - 1) - index);
+        }
+        elements[size -= 1] = null;
+    }
+
+    private void checkIndex(int index) {
```

Comment:
The `checkIndex` method should include the check for `index == size`, because calling `add` with `index == size` is a valid use case for appending to the end of the list.